### ISSUE TITLE
Overlapping events in month view

### ISSUE THREAD

**Issue by wseow87** [acro5piano/react-native-big-calendar#893]
Hi @acro5piano ,

I came across some issues with overlapping multiple days events in month view.
Given the following examples:
```typescript
var sampleEvents: Array<ICalendarEventBase> = [
  {
    title: 'Event 1',
    start: new Date(2023, 0, 1, 15, 0, 0, 0),
    end: new Date(2023, 0, 2, 10, 0, 0, 0),
  },
  {
    title: 'Event 2',
    start: new Date(2023, 0, 2, 15, 0, 0, 0),
    end: new Date(2023, 0, 3, 10, 0, 0, 0),
  },
  {
    title: 'Event 3',
    start: new Date(2023, 0, 3, 15, 0, 0, 0),
    end: new Date(2023, 0, 4, 10, 0, 0, 0),
  },
];

return (<View><Calendar mode="month" height={750} date={sampleEvents[0].start} events={sampleEvents} /></View>);
```

The calendar month view is showing as such:
![image](https://github.com/acro5piano/react-native-big-calendar/assets/82821523/6b91ea3c-5bf8-4f27-bdd3-e1cfe4af1695)

There are two issues:
1. Event 1, 2, 3 span across two days each but they are visualized as single day event.
2. Event 3 should be displayed one level lower since it should give way to Event 2 on Tuesday.

The desired output should be as such:
![image](https://github.com/acro5piano/react-native-big-calendar/assets/82821523/7ee53d18-62e4-4096-9ffa-55765e6a3962)

I created a pull request for your review:
https://github.com/acro5piano/react-native-big-calendar/pull/892

**Comment by wseow87** [comment]
I improve the sorting logic further, to optimize the empty spaces in the calendar cell.

with the given events
```typescript
var sampleEvents: Array<ICalendarEventBase> = [
  {
    title: 'Event 1',
    start: new Date(2023, 0, 1, 15, 0, 0, 0),
    end: new Date(2023, 0, 2, 10, 0, 0, 0),
  },
  {
    title: 'Event 2',
    start: new Date(2023, 0, 2, 15, 0, 0, 0),
    end: new Date(2023, 0, 4, 10, 0, 0, 0),
  },
  {
    title: 'Event 3',
    start: new Date(2023, 0, 3, 15, 0, 0, 0),
    end: new Date(2023, 0, 4, 10, 0, 0, 0),
  },
  {
    title: 'Event 4',
    start: new Date(2023, 0, 4, 15, 0, 0, 0),
    end: new Date(2023, 0, 5, 10, 0, 0, 0),
  },
  {
    title: 'Event 5',
    start: new Date(2023, 0, 5, 15, 0, 0, 0),
    end: new Date(2023, 0, 6, 10, 0, 0, 0),
  },
  {
    title: 'Event 6',
    start: new Date(2023, 0, 6, 15, 0, 0, 0),
    end: new Date(2023, 0, 7, 10, 0, 0, 0),
  },
];
```

we now get this result
![image](https://github.com/acro5piano/react-native-big-calendar/assets/82821523/11a45c68-8570-4ded-acad-ec00efed1d4f)

**Comment by acro5piano** [comment]
@wseow87 Thanks, it was released on v4.0.0!

https://github.com/acro5piano/react-native-big-calendar/releases/tag/v4.0.0
